cmake/ccache.cmake

```diff
@@ -0,0 +1,93 @@
+#[[
+Copyright 2023 Antony Peacock
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+
+include_guard(GLOBAL)
+
+
+#[=======================================================================[.rst:
+enable_ccacche
+------------------
+
+Overview
+^^^^^^^^
+
+Enable use of Ccache for compiler caching during the build process.
+
+.. code-block:: cmake
+
+  enable_ccacche(
+      [QUIET]
+  )
+   -- Searches for the ccache executable, and if found enables it for supported
+   language with specified options.
+
+  ``QUIET`` Silent output from this method.
+
+#]=======================================================================]
+function(enable_ccache)
+    set(options QUIET)
+    set(oneValueArgs)
+    set(multiValueArgs)
+    cmake_parse_arguments(CCACHE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    find_program(CCACHE_BIN ccache REQUIRED)
+    if(NOT CCACHE_BIN)
+        message(STATUS "Morpheus: CCache not found. Rebuilding all the source files.")
```

Comment:
Actually these messages are not needed, because of the `REQUIRED` parameter to `find_program`.  If `ccache` is not found CMake processing will stop with an error message at that point.  So I don't believe these messages could ever be triggered, see docs for `find_program` with the `REQUIRED` param: https://cmake.org/cmake/help/latest/command/find_program.html
```suggestion
```

Reply:
You are right, message will not be triggered because of the `REQUIRED` parameter. In this case I think that we should remove it, because for `ccache` in `conanfile.py` we are exposing an option `build_with_ccache`. Then it can effectively happen that the message will be triggered right?